### PR TITLE
Suggested slight reorg/edit for 'Processing Results'

### PR DESCRIPTION
**Pull request by ashander**
Per #349.

The docstring on variants made sense. I thought the organization of the section could be clearer and so suggested these edits

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -139,83 +139,69 @@ Variable recombination rates
...
+tree in this sequence. 
+These classes are the interfaces used to interact with the trees
+and mutational information stored in a tree sequence returned from a simulation.
+There are also methods for loading data into these objects, either from the native
+format, e.g., :func:`msprime.load`, or from another format, e.g.,
+:func:`msprime.load_text` compatible with the `Tables API <sec_tables_api>`.
+
++++++++++++++++++
+Top level-classes
++++++++++++++++++
```
[the remark this answers is not in the document]
Comment (jeromekelleher) [reply docs/api.rst]: It's for sidebar navigation really. This is to make it quick to navigate in to the TreeSequence and SparseTree classes from the top.

File: docs/api.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _sec_api:
+.. _sec-api:
```
Comment (jeromekelleher) [inline docs/api.rst]: Why the change here? I changed the separators to be underscores a while back so they will autocomplete easily in vim.

File: docs/api.rst [hunk 3]
```diff
@@ -139,83 +139,69 @@ Variable recombination rates
...
 
 +++++++++
 Constants
 +++++++++
 
-.. data:: msprime.NULL_NODE = -1
+These constants are used in :class:`.TreeSequence` and :class:`.SparseTree`
+instances to define special values of various variables.
+
+.. data:: msprime.NULL_NODE == -1
```
Comment (jeromekelleher) [inline docs/api.rst]: What's the difference with == here?

File: docs/api.rst (hunk 2 above)
Comment (ashander) [reply docs/api.rst]: Oops. That's an error! Fixed

File: docs/api.rst (hunk 3 above)
Comment (ashander) [reply docs/api.rst]: I changed to == because the single = looked like assignment, to me. With equality I can paste it into an interpreter and see 

```
In [4]: msprime.NULL_NODE == 1
Out[4]: True
```
(Of course, I could paste single = in too, but we want to discourage changing these manually in an imported module ;)

Happy to change back if it reads weirdly.

**Comment by ashander** [comment]
Great. CI failed due to that unintended label change so should be good to go pending =/== resolution. Let me know about that and I'll change if needed and squash these

**Comment by jeromekelleher** [comment]
Let's keep it as == if you think it reads better (I don't have a strong opinion).  We're good to merge after squashing.
